mesa_expense_autopopulate/models/hr/expense.py

```diff
@@ -0,0 +1,132 @@
+from odoo import api, fields, models
+
+
+class Expense(models.Model):
+    _inherit = "hr.expense"
+
+    x_studio_field_Evfct = fields.Many2one(
+        "project.project",
+        string="Project",
+        compute="_compute_project",
+        inverse="_inverse_project",
+    )
+    team_id = fields.Many2one(compute="_compute_team", inverse="_inverse_team")
+    warehouse_id = fields.Many2one(
+        compute="_compute_warehouse", inverse="_inverse_warehouse"
+    )
+    user_id = fields.Many2one(compute="_compute_user", inverse="_inverse_user")
+    company_id = fields.Many2one(compute="_compute_company", inverse="_inverse_company")
+    # The following studio field is only included here because the database would otherwise not let this module install.
+    x_studio_aa_update = fields.Many2one("account.analytic.account")
+
+    @api.multi
+    @api.depends("sale_order_id")
+    def _compute_project(self):
+        """
+        Compute the Expense's Project. It defaults to the Expense's Sale Order's
+        Project.
+
+        :return {NoneType}:
+        """
+        for expense_id in self:
+            expense_id.x_studio_field_Evfct = self.env["project.project"].search(
+                [("name", "=", expense_id.sale_order_id.x_studio_project)], limit=1
+            )
```

Comment:
Question: Is this supposed to change the sale order field's value or just be for this expense record?

```suggestion
            project_name = expense_id.sale_order_id.x_studio_project
            project_id = self.env["project.project"].search([("name", "=", project_name)], limit=1)
            
            expense_id.x_studio_field_Evfct = expense_id.force_project_id.id or project_id.id
```